src/Ball.js

```diff
@@ -44,7 +47,7 @@ export class Ball {
           return
         }
       }
-      if (this.hasAppInstalled) {
+      if (this.hasAppInstalled && this.willRespectAutoIsolation) {
```

Comment:
I think this if block must be removed:
it is quite similar to the one on line 113, with the difference that line 113 also checks for `otherBall.hasAppInstalled` (i.e. if both have the app installed).

Reply:
I agree

Reply:
Fixed with the last commit